protocol/0078-NWLI-network_wide_limits.md

```diff
@@ -0,0 +1,128 @@
+If it gets decreased below the current total party count in any given market then no further actions are needed. Specifically: all parties in that market remain unaffected (their positions and orders remain open), but no new parties are allowed to participate in the market until the total party count drops below the new limit.
+
+## Limit orders on a book
+
+Introduce a new network parameter `limits.markets.maxLimitOrders` controlling the maximum number of limit orders that can rest on a book in any given market.\
+Default value: `1,000,000`.
+
+A limit order of arbitrary volume which gets placed on the book (doesn't trade in full on entry) contributes +1 to the count. If an order already on the book gets cancelled or filled in full (so that it's remaining size is 0) the count should be decreased by 1. If the count reaches `limits.markets.maxLimitOrders` limit orders can still be submitted, but:
+
+* if the order is aggressive (results in a trade) the trade proceeds as normal,
+* if the order or its part is passive (it doesn't match in full on entry to the matching engine) and it's part would get added to the order book (it's a persistent order) that part of the order gets rejected.
+
+Pegged orders and [liquidity provision orders](./0038-OLIQ-liquidity_provision_order_type.md) do not get counted towards the limit.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.\
+If it gets decreased below the current total number of orders on the book in any given market then no further actions are needed. Specifically: all open orders in that market remain unaffected, but no new orders are allowed until the **total open order count** drops below the new limit.
+
+## Pegged orders on a market
+
+Introduce a new network parameter `limits.markets.maxPeggedOrders` controlling the maximum number of pegged orders that can be active in any given market.\
+Default value: `10,000`.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.\
+If it gets decreased below the current total number of orders on the book in any given market then no further actions are needed. Specifically: all pegged orders already present in that market remain unaffected, but no new pegged orders are allowed until the **total pegged order count** drops below the new limit.
+
+[Liquidity provision orders](./0038-OLIQ-liquidity_provision_order_type.md) do not get counted towards the limit.
+
+## LP order shapes
+
+Each [LP order shape](./0038-OLIQ-liquidity_provision_order_type.md#how-they-are-submitted) has a hardcoded limit of at most 20 entries (offsets).
+
+## LPs in a market
+
+Introduce a new network parameter `limits.markets.maxLPs` controlling the maximum number of liquidity providers that can be active in any market.\
+Default value: `1000`.
```

Comment:
I suspect this is too high because it is somehow potentially equivalent to `1 000 x 20 x 2 = 40k` pegged orders but let's wait for @peterbarrow 's comment. Even if we don't resolve this my suggestion would be to pick a default of `50` so that the PR can be merged.

Reply:
Ok, changed to 50 now.